### ISSUE TITLE
[css-env-1] Maximum safe area inset values to allow sliding bottom bar

### ISSUE THREAD

**Issue by flackr** [w3c/csswg-drafts#11019]
On some platforms, the safe-area inset can change dynamically while scrolling. For example, when the browser viewport extends to the bottom of an Android or iOS mobile screen, there is usually a system bar drawn on top of the content which also consumes input events which would prevent users from being able to read / interact with that content.

Sites creating bottom bars use the safe area to avoid overlapping this. However, if this requires a relayout it can result in a jittery experience, (e.g. see the linked demo on the [chromium intent](https://groups.google.com/a/chromium.org/g/blink-dev/c/hXkimpjYaMg/m/qiPFAW92AwAJ))

I created a demo site to show an example of two approaches times two positioning methods developers can use to create bottom bars that avoid this area:
https://flackr.github.io/web-demos/css-env/safe-area-inset/index.html

Using the `bottom` property to avoid the area in theory allows the browser to apply this as a translation similar to static sticky / fixed position offsets:
```css
.footer {
  position: fixed;
  background: gray;
  bottom: env(safe-area-inset-bottom, 0px);
}
```

However, this means that when the safe area is greater than 0, you will be able to see behind the footer's background. Using padding would grow the footer, however means that updating the footer is no longer an easily compositable effect. I think it would be useful to give developers the maximum safe area inset so that they could size the footer to the maximum size so that it can simply slide up when needed, e.g.:

```css
.footer {
  position: fixed;
  background: gray;
  padding-bottom: env(safe-area-max-inset-bottom, 0px);
  bottom: calc(env(safe-area-inset-bottom, 0px) - env(safe-area-max-inset-bottom, 0px));
}
```

This would ensure that there is extra space reserved in the footer, positioned offscreen when not needed. The linked [demo](https://flackr.github.io/web-demos/css-env/safe-area-inset/index.html) shows this approach using an assumed oversized maximum inset.

TLDR; can/should we add the maximum dynamic inset so developers can easily create a bottom bar just large enough to slide up when needed?

**Comment by fantasai** [comment]
@flackr Would it make sense to redefine the current safe area insets to be the maximum safe area inset, rather than introducing a new set of variables for it?

(And then assuming we do that, do we still need a dynamic version of these variables?)

**Comment by flackr** [comment]
@fantasai I think that only having the extra padding when it is necessary to avoid overlap with system UI is the behavior that most developers want. Having your footer always have extra space when the viewport is small so that it is large enough for when the viewport grows would likely be undesired for most sites.

These variables would both be recommended to be used for new content to provide an efficient bottom-bar.

**Comment by bramus** [comment]
> [@flackr](https://github.com/flackr) Would it make sense to redefine the current safe area insets to be the maximum safe area inset, rather than introducing a new set of variables for it, so that it affects existing content?

That would be a breaking change and defeat the purpose of these variables.

I took Rob’s proposal [for a spin](https://cdpn.io/pen/debug/wBwExjB/e642977569751476fbc6a85eed301270) and while it did take a while to click – as in: the footer gets taller but it all works out because you pull it back down – it’s a nice solution that would allow a translate-only adjustment by UAs as the viewport grows/shrinks.

_(Elements that use `padding-bottom: env(safe-area-inset-bottom, 0px);` would not get the benefits, which could be part of DevRel guidance around this)_
